backend/xivdpscalc/pps/sch.py

```diff
@@ -12,6 +17,71 @@ class SchPps(HealerPps):
     bio_potency = 70
     ed_potency = 100
 
+    def get_total_potency_variable_time(self, sim_length: ElapsedTime, character_stats: Character, rotation: SchRotation, caster_tax: float = 0.1) -> float:
+        """
+        Get the total potency in sim_length seconds
+        rotation is a SchRotation object
+        todo: support dot damage calculation
```

Comment:
i think pull out dot damage calculations into a separate function. something along the lines of:

given a rotation, identify intervals of dot uptime. calculate the strength of a dot tick for each interval in the list, multiply by number of ticks, sum it up. 

reason i suggest this is because at some point we will want to account for buff snapshotting, and also with the 25 gcd cursed sch rotation, there is brief falloff of the dot debuff. so tracking intervals should handle both cases.